### PR TITLE
Validate Create Submission Inputs

### PR DESCRIPTION
**Pull request by evan-desu**
Part of #339 

# What changed
Added validation for createSubmissions that:
- Prevents creating submissions with empty string for `healthcareProfessionalName`, `googleMapsUrl`.
- Prevents creating submissions with special characters numbers for `healthcareProfessionalName`.
- Limits `healthcareProfessionalName` to 128 characters and `googleMapsUrl` to 1028 characters.
- Prevents the use of `<script>` tags in `healthcareProfessionalName`, `googleMapsUrl`.
- Requires at least one spoken language in the `spokenLanguages` array.

# Testing instructions
Run the mutation:

```
mutation Mutation($input: CreateSubmissionInput!) {
  createSubmission(input: $input) {
    id
    googleMapsUrl
    healthcareProfessionalName
    spokenLanguages
    createdDate
  }
}
```
And the variables:
```
{
  "input": {
    "googleMapsUrl": "https://maps.google.com/exampleA",
    "healthcareProfessionalName": "",
    "spokenLanguages": [
      "en_US",
      "ja_JP"
    ]
  }
}
```
```
{
  "input": {
    "googleMapsUrl": "https://maps.google.com/exampleA",
    "healthcareProfessionalName": " ",
    "spokenLanguages": [
      "en_US",
      "ja_JP"
    ]
  }
}
```
```
{
  "input": {
    "googleMapsUrl": "https://maps.google.com/exampleA",
    "healthcareProfessionalName": "<script>Dr Joey Joe Joe</script> ",
    "spokenLanguages": [
      "en_US",
      "ja_JP"
    ]
  }
}
```
```
{
  "input": {
    "googleMapsUrl": "https://maps.google.com/exampleA",
    "healthcareProfessionalName": "Dr Joey Joe Joe",
    "spokenLanguages": [
      "en_US",
      "ja_JP"
    ]
  }
}
```
The first three will fail; the fourth will pass.

### CODE REVIEW

File: src/services/submissionService.ts [hunk 1]
```diff
@@ -161,22 +163,19 @@ export async function searchSubmissions(filters: gqlTypes.SubmissionSearchFilter
...
-            return {
+            return { 
                 data: {} as gqlTypes.Submission,
                 hasErrors: true,
-                errors: spokenLanguagesResult.errors
+                errors: validationResults.errors
             }
         }
         
+        //submissionInput.healthcareProfessionalName = submissionInput.healthcareProfessionalName?.toLowerCase()
```
Comment (ermish) [inline src/services/submissionService.ts:176]: remove comments.

File: src/validation/validateSubmissions.ts [hunk 2]
```diff
@@ -106,70 +106,72 @@ function validateSubmissionSearchBySpokenLanguage(
...
-function validateSearchByName(
+function validateSubmissionName(
     filters: gqlTypes.SubmissionSearchFilters,
-    validateSearchResults: Result<unknown>
+    validateNameResults: Result<unknown>
 ): void {
     const name = filters.healthcareProfessionalName
 
-    if (!name) {
+    if (name === undefined) {
```
Comment (ermish) [inline src/validation/validateSubmissions.ts:115]: I figured out why you did this and it makes sense, but wasn't intuitive at first. Let's add a comment here to explain why undefined here, then null later

**Review by ermish — CHANGES_REQUESTED** [review]
looks good! easy changes

File: src/validation/validateSubmissions.ts (hunk 2 above)
Comment (evan-desu) [reply src/validation/validateSubmissions.ts:117]: @ermish I added some comments. Let me know what you think.

Comment (ermish) [reply src/validation/validateSubmissions.ts:117]: looks good 👍

**Review by ermish — APPROVED** [review]
LGTM!

**Pull request merged**